src/lib/utils/murmur_hash.hpp

```diff
@@ -10,7 +10,12 @@ unsigned int murmur_hash2(const void* key, int len, unsigned int seed);
 // murmur hash for builtin number types (int, double)
 template <typename T>
 typename std::enable_if<std::is_arithmetic<T>::value, unsigned int>::type murmur2(T key, unsigned int seed) {
-  return murmur_hash2(&key, sizeof(T), seed);
+  /* 
+   * Convert all numerics to the highest precision type (double).
+   * This will cause the numbers 5 and 5.0 to have the same hash value.
+   */
+  auto double_value = static_cast<double>(key);
```

Comment:
> @mrzzzrm  mmh, auto double_value = static_cast(key); might loose data for big ints.

That is a good point. An alternative might be to move this logic into the `JoinHash` and make a few more assertions about the compatibilities of different types

Reply:
The loss of precision seems irrelevant to me. As long as the left and the right side have the same loss of precision and end up in the same bucket, this should not be an issue.